doc/classes/EditorPlugin.xml

```diff
@@ -754,5 +754,15 @@
 		<constant name="DOCK_SLOT_MAX" value="8" enum="DockSlot">
 			Represents the size of the [enum DockSlot] enum.
 		</constant>
+		<constant name="AFTER_GUI_INPUT_PASS" value="0" enum="AfterGUIInput">
+			Forwards the [InputEvent] to other EditorPlugins.
+		</constant>
+		<constant name="AFTER_GUI_INPUT_STOP" value="1" enum="AfterGUIInput">
+			Prevents the [InputEvent] from reaching other Editor classes.
+		</constant>
+		<constant name="AFTER_GUI_INPUT_EXCLUSIVE" value="2" enum="AfterGUIInput">
+			Forwards the [InputEvent] to other EditorPlugins.
+			If you use subgizmo system of [code]Node3DEditor[/code], this enum allows you to bypass normal selection behavior in [code]Node3DEditor[/code].
+		</constant>
```

Comment:
@tinmanjuggernaut @akien-mga Okay, please check this explanation whether it is clear or not.

Reply:
I am not sure if the word "exclusive" is a good.
This is just to avoid conflicts between Node3DEditorPlugin and other EditorPlugin on viewport, and I think it may conflict if there are multiple EditorPlugins that are not Node3DEditorPlugin on the viewport at the same time.

Reply:
Would "AFTER_GUI_INPUT_SUBGIZMO" be better to clarify the use case?

Reply:
1. GUI really means `Control` classes. But the manual says when "an [InputEvent] happens in the 3D viewport". So shouldn't this enum be called AFTER_VIEWPORT_INPUT....?

> If you keep the normal selection logic in a specific edit mode (PASS), you will deselect the Node you are editing when you select an empty space. However with STOP, the selection logic that use intersect_ray() will not work.
I remember this was necessary to bypass the selection process for the Nodes in the scene while still allowing the intersecting ray on viewport. It might be better to name it something else.

2. Unfortunately, I still don't understand what the code actually does. Let's build a table of events and maybe you can help explain the functions. Then we can come up with a clear english description. I'll call this mode EXCLUSIVE for now.

Based on the description, the whole purpose of this enum is to describe what happens to other EditorPlugins once we click in the viewport. So here's a guess on what happens based on what is quoted above. Can you correct  it?

| Mode | What is clicked on | My EP behavior | Other EPs behavior |
|-------|----------------------|-----------| -----------|
| PASS | Empty space | receives InputEvent, deselects node | receives InputEvent, deselects node |
| PASS | Object | receives InputEvent, selects node | receives InputEvent, selects node |
| PASS | Gizmo | receives InputEvent, manipulates gizmo | receives InputEvent, manipulates gizmo |
| STOP | Empty space | receives InputEvent, no viewport change | ignores |
| STOP | Object | receives InputEvent, no viewport change | ignores |
| PASS | Gizmo | receives InputEvent, no viewport change | ignores |
| EXCLUSIVE | Empty space | receives InputEvent, deselects node | ignores |
| EXCLUSIVE | Object | receives InputEvent, selects node | ignores |
| EXCLUSIVE | Gizmo | receives InputEvent, selects node | ignores |

If this isn't correct, then what actually happens in the third mode? The other enum entries say what happens to the InputEvent. How is the InputEvent handeled differently?

3. Where did the word "notification" come from? Is this option still handling the InputEvent or does the behavior dramatically shift from InputEvents (`_input(event)`) to Notifications (`_notification()`)? If not, there should be no mention of "notification" in any of the descriptions.

4. I have nother notes about the words Bypass, exclusive, gizmo but first lets understand the functionality. Then we'll come up with the right english words.

Reply:
@tinmanjuggernaut Probably if you don't use subgizmo, you can't understand what it does. subgizmo is a new feature implementated by @JFonS in Godot 4.0 that allows you to reuse the Node3DGizmo. For example, you can display gizmos on the vertices of a 3D object if you make plugin such as MeshEditorPlugin.

The Node3DEditor subgizmo system has function like `_subgizmos_intersect_ray()` which is called by GUIInput, but you need to handle them independently in each EditorPlugin. I called this process "notifications", but that is probably bad.

In Node3DEditor, when you select an object or blank space in the viewport, there is a behavior to select or deselect them, but this normal selection behavior gets in the way when using subgizmo.

For example, if you touch another object while in Blender's Edit mode, you will leave Edit mode and enter Object mode. To prevent this, the normal selection behavior must be bypassed.

If there are multiple EditorPlugins, `_subgizmos_intersect_ray()` will be called by all of them. At that point I worried about using the word "exclusive". So I think maybe the best name would be `AFTER_GUI_INPUT_SUBGIZMO`.

Reply:
> GUI really means Control classes. But the manual says when "an [InputEvent] happens in the 3D viewport". So shouldn't this enum be called AFTER_VIEWPORT_INPUT....?

I think it's debatable, but I want you to suggest it as another PR later.

Reply:
@JFonS I remember vaguely that I implemented it to deselect bones selected in SkeletonEditor. However, the actual behavior is to avoid conflicts with Node3DEditor.

If anyone implements some EditorPlugins, whether it does deselection or not depends on the behavior of the EditorPlugin, so I think it is better to use a name that describes that behavior rather than a specific purpose.

Reply:
So what is the current behavior? What changes between `DESELECT` and `STOP`?

Reply:
@JFonS As I wrote above https://github.com/godotengine/godot/pull/64465#discussion_r963152072, it just avoid selecting in Node3DEditor.

Reply:
I'm really not familiar with all this, but if it's something which has implementation-specific behavior, maybe it should be named something like `AFTER_GUI_INPUT_CUSTOM`? If it's meant to be used as a "don't stop and don't pass, code the behavior yourself" kind of flag.

Reply:
Alright, then I would go with something like `AFTER_GUI_INPUT_PASS_AND_KEEP_SELECTION` and the docs "Pass the input event to other editor plugins but prevent the editor from changing selection."

Reply:
Renamed it to `AFTER_GUI_INPUT_CUSTOM ` for now.
The current description in the documentation should be valid.
InputEvent is forwarded from Node3DEditor to each EditorPlugin, but is is not processed by Node3DEditor itself such a case for selection behavior.